api/provider/aws/service_read.go

```diff
@@ -77,13 +80,38 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 	})
 
 	output, err := s.AWS.ECS.DescribeServices(input)
-	if err != nil || len(output.Services) == 0 {
-		return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist")
+	if err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "ServiceNotFoundException" {
+			return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
+		}
+
+		return nil, err
+	}
+
+	if len(output.Services) == 0 {
+		return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
 	}
 
 	return output.Services[0], nil
 }
 
+func (s *ServiceProvider) lookupDeployIDFromTaskDefinitionARN(taskDefinitionARN string) (string, error) {
+	tags, err := s.TagStore.SelectByType("deploy")
+	if err != nil {
+		return "", err
+	}
+
+	if len(tags) == 0 {
+		return "", errors.Newf(errors.DeployDoesNotExist, "No deploys exist")
+	}
+
+	if tag, ok := tags.WithValue(taskDefinitionARN).First(); ok {
```

Comment:
should be `tags.WithKey("arn").WithValue(taskDefinitionARN).First()`

Reply:
✅